JH2 - 5/10/2016

olegbaslak opened the pull request:
done lesson1.task04

olegbaslak committed:
CI and maven init

olegbaslak committed:
CI and maven init

olegbaslak committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	.travis.yml

olegbaslak committed:
Delete .pom

olegbaslak committed:
lesson1.task04

olegbaslak committed:
lesson1.task04 done

olegbaslak committed:
lesson1.task05 init structure

olegbaslak committed:
lesson1.task05 exceptions added

olegbaslak committed:
small refactoring (public classes changed to package-private)

olegbaslak committed:
lesson1task03 bug fix with zero check

olegbaslak committed:
lesson1task05 triangle class

olegbaslak committed:
lesson1task04 fix

olegbaslak committed:
lesson1task05 done

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/Validator.java:
```diff
@@ -0,0 +1,38 @@
+
+class Validator {
+
+    static double[] parse(String[] args) {
```
Missed comment for method parse

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/Validator.java:
```diff
@@ -0,0 +1,38 @@
+            for (int i = 0; i < parsedCoefficients.length; i++) {
+                parsedCoefficients[i] = Double.parseDouble(args[i]);
+            }
+        }catch (NumberFormatException e){
```
Missed spaces

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/Validator.java:
```diff
@@ -0,0 +1,38 @@
+                parsedCoefficients[i] = Double.parseDouble(args[i]);
+            }
+        }catch (NumberFormatException e){
+            e.getMessage();
```
Unused return value of statement e.getMessage();

olegbaslak replied:
Check new commit

AlexanderSoroka reviewed (changes requested):
It looks quite good but there are some issues to be fixed

olegbaslak committed:
lesson1task04 validator fixes:
-doc added
-refactored
-exception handling
version 1.1

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/Main.java:
```diff
@@ -0,0 +1,42 @@
+        }
+
+        try {
+            SquareEquation squareEquation = new SquareEquation(Validator.parse(args));
```
Is zero acceptable value for paramA? I don't see this check

olegbaslak replied:
Oooops

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/Validator.java:
```diff
@@ -0,0 +1,48 @@
+            }
+        } catch (NumberFormatException e) {
+            System.out.println(PARSE_ERROR);
+
```
Useless blank line

AlexanderSoroka commented on src/com/rct/homework/lesson1/task05/Main.java:
```diff
@@ -0,0 +1,43 @@
+     *                                    don't exist
+     */
+    public static void main(String[] args) {
+
```
Useless blank line

AlexanderSoroka commented on src/com/rct/homework/lesson1/task05/Main.java:
```diff
@@ -0,0 +1,43 @@
+    public static void main(String[] args) {
+
+        System.out.println(INPUT_REQUSET);
+
```
Useless blank line

olegbaslak replied:
I tried to divide code to logical blocks

olegbaslak committed:
lesson1task05 refactoring

olegbaslak committed:
lesson1task04 refactoring

olegbaslak committed:
lesson1task05 fix
correct handling variant when some of params are equal zero

olegbaslak committed:
lesson1task05 doc updated
app version 1.1

Gpef commented:
Check new commits please

AlexanderSoroka commented on src/com/rct/homework/lesson1/task05/InputController.java:
```diff
@@ -0,0 +1,61 @@
+    private static final String TRY_AGAIN_MESSAGE = "Try again";
+
+    /**
+     * Gets {@code} paramsNumber arguments from console input.
```
This doesn;t say that the method ignores negatives

AlexanderSoroka commented on src/com/rct/homework/lesson1/task05/InputController.java:
```diff
@@ -0,0 +1,61 @@
+        int index = 0;
+        while (index < paramsNumber) {
+            try {
+                readString = scanner.next();
```
What if user inputs something very big like 2e2048?

olegbaslak committed:
lesson1task05 update
-fixed too big number bug
-fixed negative number of arguments bug
ver. 1.2

Gpef commented:
Fresh updates incoming

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/SquareEquation.java:
```diff
@@ -0,0 +1,97 @@
+    private double findDiscriminant() {
+        double d;
+        d = this.paramB * this.paramB - 4 * this.paramA * this.paramC;
+        return d;
```
return paramB_paramB - 4_paramA*paramC; is shorter and simpler to read, isn't it?

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/SquareEquation.java:
```diff
@@ -0,0 +1,97 @@
+    double[] findRoots() throws ComplexRootsOnlyException, NoRootsException {
+        double[] roots = new double[2];
+
+        /* eq. c = 0 */
```
Here and everywhere use // comments

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/SquareEquation.java:
```diff
@@ -0,0 +1,97 @@
+        }
+
+        /* eq. bx = -c */
+        if (Validator.isZero(paramA) && !Validator.isZero(paramB)) {
```
This is not quadratic equation

AlexanderSoroka replied:
I think most of these checks are useless

olegbaslak committed:
lesson1task04 find discriminant refactor
version 1.2

olegbaslak committed:
lesson1task04 added doc to NoRootsException class

olegbaslak committed:
lesson1task04 added NotQuadraticEquationException

olegbaslak committed:
lesson1task04 SquareEquation update
deleted unnecessary ifs
added exception throw when created equation isn't quadratic

olegbaslak committed:
lesson1task04 new support for Main
it now supports NotQuadraticEquationException

olegbaslak committed:
lesson1task04 Main refactored

olegbaslak committed:
lesson1task04 update SquareEquation
deleted unnecessary code

olegbaslak committed:
lesson1task04 NoRootsException deleted for it's uselessness

olegbaslak committed:
lesson1task04 refactored

AlexanderSoroka commented on src/com/rct/homework/lesson1/task04/Validator.java:
```diff
@@ -0,0 +1,46 @@
+            for (int i = 0; i < parsedCoefficients.length; i++) {
+                parsedCoefficients[i] = Double.parseDouble(args[i]);
+            }
+        } catch (NumberFormatException e) {
```
Is it better to expose it for caller and let it decides if application should stop or something else? Method that hides inside System.exit(1) is really bad idea - at least it must be well-documented

AlexanderSoroka commented on src/com/rct/homework/lesson1/task05/Main.java:
```diff
@@ -0,0 +1,48 @@
+        System.out.println(INPUT_REQUSET);
+        InputController inputController = new InputController();
+        double[] sides = new double[0];
+        try {
```
Let combine this 2 try catch to one

olegbaslak committed:
lesson1task05 refactored
two try-catch block united into one

olegbaslak committed:
lesson1task04 parse error throw to parse method caller
to let the caller decide what to do with exception

olegbaslak committed:
lesson1task04 refactored printing results
using String.format instead of concatenation of Strings

olegbaslak committed:
lesson1task05 refactored
- deleted useless code
- other optimizations

olegbaslak committed:
lesson1task05 refactored

olegbaslak committed:
lesson1task05 nothing important
-refactoring
-deleted unnecessary code

olegbaslak committed:
lesson1task05 nothing important
-refactoring
